kotlinx-coroutines-core/common/src/channels/ConflatedChannel.kt

```diff
@@ -54,7 +49,7 @@ internal open class ConflatedChannel<E> : AbstractChannel<E>() {
                     }
                 }
             }
-            value = element
+            updateValueLocked(element)?.let { throw it }
```

Comment:
Regular channels report these exceptions to global exception handler, while this one rethrows an exception.
Let's stick to a consistent approach

Reply:
There's no global exception handler to report the exception to here. Channel has no context on its own. In other places we have an inverse situation. E.g., during resume we cannot just rethow exception (since execution stack is arbitrary and unrelated to the issue), but we have a context with an exception handler on our hands.